packages/dbml-core/src/export/DbmlExporter.js

```diff
@@ -263,59 +266,67 @@ class DbmlExporter {
     return strArr.length ? strArr.join('\n') : '';
   }
 
+  static getTableGroupSettings (tableGroup) {
+    const settings = [];
+
+    if (tableGroup.color) settings.push(`color: ${tableGroup.color}`);
+
+    return settings.length ? ` [${settings.join(', ')}]` : '';
+  }
+
   static exportTableGroups (tableGroupIds, model) {
     const tableGroupStrs = tableGroupIds.map(groupId => {
       const group = model.tableGroups[groupId];
       const groupSchema = model.schemas[group.schemaId];
+      const groupSettingStr = DbmlExporter.getTableGroupSettings(group);
 
-      return `TableGroup ${shouldPrintSchema(groupSchema, model)
-        ? `"${groupSchema.name}".` : ''}"${group.name}" {\n${
-        group.tableIds.map(tableId => {
+      const groupNote = group.note ? `  Note: ${DbmlExporter.escapeNote(group.note)}\n`: '';
+      const groupSchemaName = shouldPrintSchema(groupSchema, model) ? `"${groupSchema.name}".` : '';
+      const groupName = `${groupSchemaName}"${group.name}"`;
+
+      const tableStr = group.tableIds
+        .map(tableId => {
           const table = model.tables[tableId];
           const tableSchema = model.schemas[table.schemaId];
           return `  ${shouldPrintSchema(tableSchema, model)
-            ? `"${tableSchema.name}".` : ''}"${table.name}"`;
-        }).join('\n')}\n}\n`;
+            ? `"${tableSchema.name}".` : ''}"${table.name}"\n`;
+        })
+        .join('');
+
+      return `TableGroup ${groupName}${groupSettingStr} {\n${tableStr}${groupNote}}\n`;
     });
 
     return tableGroupStrs.length ? tableGroupStrs.join('\n') : '';
   }
 
+  static exportStickyNotes (model) {
+    const noteStrs = [];
+    forIn(model.notes, (note) => {
+      const escapedContent = `  ${DbmlExporter.escapeNote(note.content)}`;
+      noteStrs.push(`Note ${note.name} {\n${escapedContent}\n}\n`);
+    });
+    return noteStrs.join('\n');
+  }
+
   static export (model) {
-    let res = '';
-    let hasBlockAbove = false;
+    const elementStrs = [];
     const database = model.database['1'];
 
     database.schemaIds.forEach((schemaId) => {
       const {
         enumIds, tableIds, tableGroupIds, refIds,
       } = model.schemas[schemaId];
 
-      if (!_.isEmpty(enumIds)) {
-        if (hasBlockAbove) res += '\n';
-        res += DbmlExporter.exportEnums(enumIds, model);
-        hasBlockAbove = true;
-      }
-
-      if (!_.isEmpty(tableIds)) {
-        if (hasBlockAbove) res += '\n';
-        res += DbmlExporter.exportTables(tableIds, model);
-        hasBlockAbove = true;
-      }
+      if (!isEmpty(enumIds)) elementStrs.push(DbmlExporter.exportEnums(enumIds, model));
+      if (!isEmpty(tableIds)) elementStrs.push(DbmlExporter.exportTables(tableIds, model));
+      if (!isEmpty(tableGroupIds)) elementStrs.push(DbmlExporter.exportTableGroups(tableGroupIds, model));
+      if (!isEmpty(refIds)) elementStrs.push(DbmlExporter.exportRefs(refIds, model));
```

Comment:
The logic here appears to be different from the previous one. Why do we need to take care of these legacy codes? Do we have tests to ensure it won't cause any issues? cc @NQPhuc @thonx-holistics

Reply:
We do have some tests for `table, enum, ref` under `dbml-core/__test__/exporter/model_exporter/dbml_exporter`. Please help add more test for `table group, sticky note` @thonx-holistics

Reply:
The logic should stay the same. Previously, before adding the DBML content of some elements to the result, the function will check if there is already an element in the DBML content to add a newline to separate them. Ultimately, the purpose of the old code is to separate DBML elements with newlines for clarity.

That's why I refactored it using the `join` method instead, to make the function more readable.

There are some unit tests to cover this, but I think they are not enough, so I'm writing some more tests.